### PR TITLE
Created declaration-block-trailing-semicolon with options

### PR DESCRIPTION
**Pull request by davidtheclark**
Changes the name of the rule, as @jeddy3 mentioned, and adds the option to either expect *or reject* the trailing semicolon -- which I noticed when I re-read that part of the rules gist.

### CODE REVIEW

File: src/rules/declaration-block-trailing-semicolon/__tests__/index.js [hunk 1]
```diff
@@ -0,0 +1,153 @@
+import test from "tape"
+import testRule from "../../__tests__/utils/testRule"
+
+import trailingSemicolon from ".."
+
+const testTrailingSemicolon = testRule(trailingSemicolon)
+const expectedError = "Expected a trailing semicolon (declaration-block-trailing-semicolon)"
+const rejectedError = "Expected no trailing semicolon (declaration-block-trailing-semicolon)"
```
Comment (MoOx) [inline src/rules/declaration-block-trailing-semicolon/__tests__/index.js]: maybe we can get those messages exported from the rule directly. With es6 we can have multiples values exported: a default one `import rule from "./rule"` and some others like `import rule, {messages} from "./rule"`. That will make maintenance easier. 
In the rule, we can just add `export const messages = { /*...*/ }`

**Comment by MoOx** [comment]
You can rebase on master so tests on appveyor pass too.

**Comment by jeddy3** [comment]
Great to see the options go in! I thought about that last night and was going to give it a shot myself today, but you beat me to it :)

**Comment by MoOx** [comment]
We should made an "official" recap for rules names. Some guidelines or something. Some simple rules to build rule names.

**Comment by jeddy3** [comment]
I can have a go that, if you like? 

Where would you like it? Just in the readme for now, or a standalone markdown file somewhere?

**Comment by MoOx** [comment]
As you want. Maybe we should start in a `docs/naming-rules.md` (+ link to this folder/toc in the README) ?

**Comment by davidtheclark** [comment]
I rebased to the 0.1.0 branch, because that had the changes appveyor needed; and I implemented the suggestion you made @MoOx which I thought was a good one.

**Comment by MoOx** [comment]
Note that you can do PR from a branch of this repo. Don't bother you to use a fork since you are a owner ;)

**Comment by davidtheclark** [comment]
Ok

**Pull request merged**